### PR TITLE
Kalliope-Import-Plugin

### PR DESCRIPTION
**Pull request by solth**
Import metadata for documents via the Kalliope SRU interface

### CODE REVIEW

File: Goobi/plugins/opac/ModsPlugin/org/kitodo/production/plugin/CataloguePlugin/ModsPlugin/UGHUtils.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
+/*
```
Comment (stweil) [inline Goobi/plugins/opac/ModsPlugin/org/kitodo/production/plugin/CataloguePlugin/ModsPlugin/UGHUtils.java]: Would it be better to add this file to Kitodo.ugh?

File: Goobi/plugins/opac/PicaPlugin/org/goobi/production/plugin/CataloguePlugin/PicaPlugin/PicaPlugin.java [hunk 2]
```diff
@@ -393,10 +411,11 @@ public static String getDescription(Locale language) {
 			throw e;
 		} catch (Exception e) {
 			throw new RuntimeException(e.getMessage(), e);
-		}
+		}		
+		
 		return createResult(cod.getTitle(), myFirstHit, ff);
 	}
-
+
```
Comment (stweil) [inline Goobi/plugins/opac/PicaPlugin/org/goobi/production/plugin/CataloguePlugin/PicaPlugin/PicaPlugin.java]: Please remove any spaces at line endings.

File: Goobi/src/org/goobi/production/plugin/CataloguePlugin/CataloguePlugin.java [hunk 3]
```diff
@@ -131,6 +139,11 @@
 public class CataloguePlugin extends UnspecificPlugin {
 
 	/**
+	 * Local log4j logger.
+	 */
+	private static final Logger logger = Logger.getLogger(ConfigOpac.class);
```
Comment (henning-gerhardt) [inline Goobi/src/org/goobi/production/plugin/CataloguePlugin/CataloguePlugin.java]: Class name is `CataloguePlugin` and logger should be identical.

File: Goobi/src/de/sub/goobi/forms/ProzesskopieForm.java [hunk 4]
```diff
@@ -1573,6 +1669,14 @@ public void setOpacSuchbegriff(String opacSuchbegriff) {
 		this.opacSuchbegriff = opacSuchbegriff;
 	}
 
+	public String getInstitution() {
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/forms/ProzesskopieForm.java]: A short JavaDoc entry should be nice for this getter Method.

File: Goobi/src/de/sub/goobi/forms/ProzesskopieForm.java [hunk 5]
```diff
@@ -1573,6 +1669,14 @@ public void setOpacSuchbegriff(String opacSuchbegriff) {
...
+		return institution;
+	}
+
+	public void setInstitution(String institution) {
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/forms/ProzesskopieForm.java]: A short JavaDoc entry should be nice for this stter Method.

File: Goobi/plugins/opac/ModsPlugin/org/kitodo/production/plugin/CataloguePlugin/ModsPlugin/package-info.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ * org.goobi.production.plugin.CataloguePlugin.CataloguePlugin.
+ *
+ * Most of the code originates form the package <kbd>de.unigoettingen.sub.
+ * search.opac</kbd> created by “Ludwig” and maybe others in 2005. The plugin
+ * classes can be compiled with the libraries from /Goobi/WEB-INF/lib and the
+ * Tomcat server runtime libraries on the classpath.
+ *
+ * @author Arved Solth, Christopher Timm
+ */
+package org.kitodo.production.plugin.CataloguePlugin.ModsPlugin;
```
Comment (henning-gerhardt) [inline Goobi/plugins/opac/ModsPlugin/org/kitodo/production/plugin/CataloguePlugin/ModsPlugin/package-info.java]: Line break at end of file would be nice.

File: Goobi/plugins/opac/ModsPlugin/org/kitodo/production/plugin/CataloguePlugin/ModsPlugin/kalliope2kitodo.xsl [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    <xsl:template match="mods:mods/mods:location/mods:url">
+        <goobi:metadata name="slub_link"><xsl:value-of select="normalize-space()" /></goobi:metadata>
+        <goobi:metadata name="slub_linktext">Katalognachweis</goobi:metadata>
+    </xsl:template>
+
+    <!-- pass-through rule -->
+    <xsl:template match="@*|node()">
+        <xsl:apply-templates select="@*|node()"/>
+    </xsl:template>
+</xsl:stylesheet>
```
Comment (henning-gerhardt) [inline Goobi/plugins/opac/ModsPlugin/org/kitodo/production/plugin/CataloguePlugin/ModsPlugin/kalliope2kitodo.xsl]: Line break at end of file would be nice.

File: Goobi/plugins/opac/ModsPlugin/org/kitodo/production/plugin/CataloguePlugin/ModsPlugin/UGHUtils.java (hunk 1 above)
Comment (henning-gerhardt) [inline Goobi/plugins/opac/ModsPlugin/org/kitodo/production/plugin/CataloguePlugin/ModsPlugin/UGHUtils.java]: As this file contains plugin specific code, this file should be stay inside plugin. There is a identical named file in PicaPlugin too.

File: Goobi/config/kitodo_mods_opac.xml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+		</filterInstitutions>
+		<additionalDetails>
+			<detail name="creator" xpath="//goobi:displayName" />
+			<detail name="title" xpath="//goobi:metadata[@name='TitleDocMain']" />
+			<detail name="shelfmarksource" xpath="//goobi:metadata[@name='shelfmarksource']" />
+			<detail name="place" xpath="//goobi:metadata[@name='PlaceOfPublication']" />
+			<detail name="url" xpath="//goobi:metadata[@name='slub_link']" />
+		</additionalDetails>
+	</catalogue>
+</opacCatalogues>
```
Comment (stweil) [inline Goobi/config/kitodo_mods_opac.xml]: Missing LF in last line.

File: Goobi/plugins/opac/PicaPlugin/org/goobi/production/plugin/CataloguePlugin/PicaPlugin/PicaPlugin.java [hunk 9]
```diff
@@ -45,6 +49,8 @@
 import org.joda.time.LocalTime;
 import org.w3c.dom.Node;
 
+import net.xeoh.plugins.base.Plugin;
```
Comment (stweil) [inline Goobi/plugins/opac/PicaPlugin/org/goobi/production/plugin/CataloguePlugin/PicaPlugin/PicaPlugin.java]: Sort import statements alphabetically?

**Comment by solth** [comment]
- added javadoc to functions "getInstitution" and "setInstitution" in class "Prozesskopieform"
- removed trailing whitespace from PicaPlugin.java
- added LF to files kalliope2kitodo.xsl, package-info.java and kitodo_mods_opac.xml 
- updated logger name in CataloguePlugin.java

**Comment by henning-gerhardt** [comment]
@stweil: is something left over which must be corrected or is PR ready to merge?

**Pull request merged**